### PR TITLE
Add Application Service registration type requirement + py35, pg95 deprecation notices to v1.32.0 upgrade notes

### PR DESCRIPTION
**Pull request by anoadragon453**
Fixes https://github.com/matrix-org/synapse/issues/9846.

Adds important removal information from the top of https://github.com/matrix-org/synapse/releases/tag/v1.32.0rc1 into UPGRADE.rst.

### CODE REVIEW

File: UPGRADE.rst [hunk 1]
```diff
@@ -88,6 +88,14 @@ for example:
 Upgrading to v1.32.0
 ====================
 
+Dropping support for end-of-life Python, Postgres and SQLite versions
+---------------------------------------------------------------------
+
+In line with our `deprecation policy <https://github.com/matrix-org/synapse/blob/v1.32.0rc1/docs/deprecation_policy.md>`_,
```
Comment (babolivier) [inline UPGRADE.rst:94]: Maybe want to change this link so it points to the stable release (or master) instead of the RC?

**Review by babolivier — APPROVED** [review]
lgtm otherwise (modulo missing newsfile)

**Comment by anoadragon453** [comment]
I'm not sure if we want a newsfile for this change? It would be the only change since the RC.

File: UPGRADE.rst (hunk 1 above)
Comment (anoadragon453) [reply UPGRADE.rst]: Probably best to keep it at the current stable just in case we change it significantly (or move it).

File: UPGRADE.rst [hunk 2]
```diff
@@ -88,6 +88,14 @@ for example:
 Upgrading to v1.32.0
 ====================
 
+Dropping support for end-of-life Python, Postgres and SQLite versions
```
Comment (callahad) [inline UPGRADE.rst:91]: This is the most tiny nitpick, but I don't think minor versions of SQLite actually go EOL / have a defined support lifetimes.

Maybe just `s/end-of-life/old/` on the heading?

**Comment by anoadragon453** [comment]
We've decided to merge without a changelog entry.

**Pull request merged**